src/services/submissionService.ts

```diff
@@ -140,22 +140,21 @@ export const addSubmission = async (submissionInput: gqlType.SubmissionInput):
             errors: spokenLanguagesResult.errors
         }
     }
+    
+    const submissionRef = dbInstance.collection('submissions').doc()
+    const newSubmissionId = submissionRef.id
 
-    const dbSubmission = convertToDbSubmission({
+    const newSubmission = convertToDbSubmission({
         ...submissionInput,
-        spokenLanguages: spokenLanguagesResult.data as gqlType.SpokenLanguageInput[]
+        spokenLanguages: spokenLanguagesResult.data as gqlType.SpokenLanguageInput[],
+        id: newSubmissionId
     })
 
-    const submissionRef = dbInstance.collection('submissions')
-
-    const docRef = await submissionRef.add(dbSubmission)
-
-    const newSubmission: dbSchema.Submission = {
-        ...dbSubmission,
-        id: docRef.id
-    }
+    await submissionRef.set(newSubmission)
 
     addSubmissionResult.data = newSubmission
+    console.log('Generated ID:', newSubmission.id)
```

Comment:
should add more detail that this is for a new submission
```suggestion
    console.log('Generated new submission ID:', newSubmission.id)
```

Reply:
@ermish Got it. Will make the change and then merge it!